src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -191,6 +193,22 @@ public override void Close()
 		}
 #endif
 
+		public override T GetFieldValue<T>(int ordinal)
+		{
+			if (typeof(T) == typeof(DateTimeOffset))
+				return (T) Convert.ChangeType(GetDateTimeOffset(ordinal), typeof(T));
+
+			return base.GetFieldValue<T>(ordinal);
+		}
+
+		public override async Task<T> GetFieldValueAsync<T>(int ordinal, CancellationToken cancellationToken)
+		{
+			if (typeof(T) == typeof(DateTimeOffset))
+				return (T) Convert.ChangeType(GetDateTimeOffset(ordinal), typeof(T));
```

Comment:
It turns out that the [base method already does this](https://github.com/dotnet/corefx/blob/release/1.0.0/src/System.Data.Common/src/System/Data/Common/DbDataReader.cs/#L199)!  The override can be eliminated, although I will keep the test in place to ensure if the base method ever changes we are still in good shape.